corehq/apps/domain/forms.py

```diff
@@ -954,12 +956,9 @@ def save(self, domain):
 pwd_pattern = re.compile( r"([-\w]){"  + str(min_pwd) + ',' + str(max_pwd) + '}' )
 
 def clean_password(txt):
-    if len(txt) < min_pwd:
-        raise forms.ValidationError('Password is too short; must be at least %s characters' % min_pwd )
-    if len(txt) > max_pwd:
-        raise forms.ValidationError('Password is too long; must be less than %s characters' % max_pwd )
-    if not pwd_pattern.match(txt):
-        raise forms.ValidationError('Password may only contain letters, numbers, hyphens, and underscores')
+    strength = zxcvbn(txt, user_inputs=['commcare', 'hq', 'dimagi', 'commcarehq'])
```

Comment:
These user_inputs are penalized in the strength algorithm, such that a password like dimagicommcare123 which would look pretty random outside of the Commcare context would not be as strong.